### ISSUE TITLE
[ENH]: Better URL support for PDF backend

### ISSUE THREAD

**Issue by oscargus** [matplotlib/matplotlib#23152]
### Problem

The SVG backend supports URLs for all objects, which is quite easy as the URL is connected to the object in SVG (if I understand things correctly).

The PDF backend supports URLs for texts, but not for any other objects. One problem here is that the link seems to be connected to an area rather than an object.

It should be reasonably feasible to add support for images, markers, and bars. 

### Proposed solution

This is the code for adding an URL to a text object:
https://github.com/matplotlib/matplotlib/blob/3df958c760dbde3a6c576fefa7827a136385b5c3/lib/matplotlib/backends/backend_pdf.py#L2156-L2167

Basically, if one knows the coordinates, it should just be to copy this.

**Comment by oscargus** [comment]
Also see #23205 that contains a link to the PDF specification and points out a problem with the current approach above.

**Comment by oscargus** [comment]
Note that in #23286 a function is added that avoids a bit of code duplication, so rather than copying the whole block one calls a function with relevant parameters that generates it. When #23205 is solved, it can also rotate the clickable area.
